apps/web/app/query/post.query.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from "@tanstack/react-query";
+import { defineQueryOptions } from "./util";
+
+const postsQueryPrefix = "posts";
+
+export const postQueryKey = {
+    id: [postsQueryPrefix, "id"],
+} as const;
+
+export const postQueryOptions = defineQueryOptions({
+    queryKey: postQueryKey.id,
+    queryFn({ api }) {
+        return api.postApi.getAllPosts();
+    },
+});
+
+export function usePostQuery() {
+    return useQuery(postQueryOptions);
+}
```

Comment:
Since you're fetching an array of posts, it would make more sense to rename this hook to `usePostsQuery`. And the `postQueryOptions` to `postsQueryOptions`

```suggestion
export function usePostsQuery() {
    return useQuery(postsQueryOptions);
}
```